src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let objectKeeper = { ...state };
+  const stateModified = [];
+
+  for (const index of actions) {
+    if (index.type === 'addProperties') {
+      const localTemporaryObject = { ...objectKeeper };
+
+      for (const key in index.extraData) {
+        localTemporaryObject[key] = index.extraData[key];
+      }
+
+      objectKeeper = { ...localTemporaryObject };
+      stateModified.push(localTemporaryObject);
```

Comment:
Check comment above about unnecessary copy.
Also, you perform push in all cases. Do it once after `switch/case` block.